llpc/tool/amdllpc.cpp

```diff
@@ -1765,6 +1767,148 @@ void FindAllMatchFiles(
     return;
 }
 #endif
+
+// =====================================================================================================================
+// Builds necessary variants of the shader inFile.  It must be either spir-v assembly or spir-v binary.
+static Result BuildShaderVariants(
+    ICompiler*            pCompiler,   // [in] LLPC context
+    const std::string& inFile)        // Input filename
+{
+    Result result = Result::Success;
+    CompileInfo compileInfo = {};
+    std::string fileNames;
+    compileInfo.doAutoLayout = true;
+    assert(!CheckAutoLayoutCompatible && "When processing shader variants we do not use the auto layout.");
+    compileInfo.checkAutoLayoutCompatible = false;
+
+    result = InitCompileInfo(&compileInfo);
+
+    //
+    // Translate sources to SPIR-V binary
+    //
+    std::string spvBinFile;
+    assert((IsSpirvTextFile(inFile) || IsSpirvBinaryFile(inFile)) && "We only build shader variants for spir-v files.");
+
+    if (IsSpirvTextFile(inFile))
+    {
+        result = AssembleSpirv(inFile, spvBinFile);
+    }
+    else
+    {
+        spvBinFile = inFile;
+    }
+
+    BinaryData spvBin = {};
+
+    if (result == Result::Success)
+    {
+        result = GetSpirvBinaryFromFile(spvBinFile, &spvBin);
+
+        if (result == Result::Success)
+        {
+            if (InitSpvGen() == false)
+            {
+                LLPC_OUTS("Failed to load SPVGEN -- no SPIR-V disassembler available\n");
+            }
+            else
+            {
+                // Disassemble SPIR-V code
+                uint32_t textSize = spvBin.codeSize * 10 + 1024;
+                char* pSpvText = new char[textSize];
+                LLPC_ASSERT(pSpvText != nullptr);
+                memset(pSpvText, 0, textSize);
```

Comment:
You can use `new char[textSize]()` to get zero-initialized memory.
Or even better, just create a vector of chars.